composeApp/src/commonMain/kotlin/com/sixbynine/transit/path/app/ui/ColorRect.kt

```diff
@@ -0,0 +1,68 @@
+package com.sixbynine.transit.path.app.ui
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.border
+import androidx.compose.foundation.isSystemInDarkTheme
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.RectangleShape
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import com.sixbynine.transit.path.model.ColorWrapper
+import com.sixbynine.transit.path.model.unwrap
+import com.sixbynine.transit.path.util.conditional
+
+@Composable
+fun ColorRect(
+    colors: List<ColorWrapper>,
+    height: Dp = 48.dp,
+    modifier: Modifier = Modifier
+) {
+    Box(
+        modifier
+            .width(12.dp)
+            .height(height)
+    ) {
+        // First color - fills entire height
+        colors.firstOrNull()?.let {
+            Box(
+                Modifier
+                    .width(12.dp)
+                    .height(height)
+                    .background(it.unwrap())
+            )
+        }
+
+        // Second color - fills bottom 50%
+        colors.getOrNull(1)?.let {
+            Box(
+                Modifier
+                    .width(12.dp)
+                    .height(height)
+                    .padding(start = 0.dp, top = height / 2)
+                    .clip(RectangleShape)
+                    .background(it.unwrap())
+            )
+        }
+
+        // Third color - fills bottom 25%
+        colors.getOrNull(2)?.let {
```

Comment:
Should we split these evenly if there are three colors? It's rare, but can happen when there are route changes.

Easiest way is probably to use a `Column` for the parent instead of a box, and give the children each the same `weight`

We could actually simplify the code a lot more by just iterating over the colors.

Like:

```kotlin
Column(...) {
    colors.take(3).forEach { color ->
        Box(
            Modifier
                .width(12.dp)
                .weight(1f)
                .background(color.unwrap())
        )
    }
}
```